dataprocspawner/spawner.py

```diff
@@ -1024,6 +1027,28 @@ def get_rand_string(self, length):
     rand_str = ''.join((random.choice(letters_and_digits) for i in range(length)))
     return rand_str
 
+  def _validate_zones(self, region, zones):
```

Comment:
General note: wherever this is called, we should catch and log exceptions so that failure to validate zones won't completely break the Hub experience. This is doubly important since listing zones requires a separate IAM permission that might not always be granted.

Reply:
OK, try/catch + logging will help us with that. Already implemented.